feat: add log viewer options

SethFalco opened the pull request:
Whenever I view the logs of a pod, I always set the same options pretty much. ^-^'
This just creates some global settings which sets the default log viewer options.

For now, it only covers `follow`, `timestamp` and `wrap`.
I'm considering doing others later, but figured for now this would be nice.

It looks to me (though I didn't check manually) that the `reset()` method wasn't resetting the `wrap` checkbox anyway, so this also fixes that.

### Related
* Partially resolves x https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/issues/1058#issue-1204480246
  * I can also make an option to run by default on viewing logs too after/if this is merged.

SethFalco commented:
While we're at it, I'm just wondering about this. 🤔 

```js
// eslint-disable-next-line @typescript-eslint/prefer-for-of
for (let i = 0; i < newContent.length; i += 1) {
    if (newContent[i].length > 0) {
        content[counter] = newContent[i];
        fpcCounter++;
        fullPageContent[fpcCounter] = newContent[i];
        counter++;
    }
}
```

Is there a reason you're not using a for-each? I normally would change this without asking, but given in the repo the warning it explicitly disabled too, I wanted to ask about it.

lstocchi commented on src/components/logs/logsWebview.ts:
```diff
@@ -126,4 +140,16 @@ export class LogsPanel extends WebPanel {
 
+    private getLogViewerOptions(): object {
+        const configuration = vscode.workspace.getConfiguration();
+        const follow = configuration.get('vscode-kubernetes.log-viewer.follow', false);
```
Better to move these to `config.ts` for simmetry and so that config remains responsible for knowing the key names.

lstocchi reviewed (commented):
Thanks for this @SethFalco. Looks good. 
I'm wondering if we should add a button or anything else to save the settings directly from the viewer. It would be faster for users to set them and also it would be easily seen by those that do not read the README/changelog. @itowlson WDYT?

lstocchi commented:
> While we're at it, I'm just wondering about this. thinking
> 
> ```js
> // eslint-disable-next-line @typescript-eslint/prefer-for-of
> for (let i = 0; i < newContent.length; i += 1) {
>     if (newContent[i].length > 0) {
>         content[counter] = newContent[i];
>         fpcCounter++;
>         fullPageContent[fpcCounter] = newContent[i];
>         counter++;
>     }
> }
> ```
> 
> Is there a reason you're not using a for-each? I normally would change this without asking prior, but given in the repo the warning is explicitly disabled too, I wanted to ask about it.

The log viewer has a long and troubled history as it changed quite a bit lately. I guess this was made during some performance enhancement PRs. First we showed all logs in a single push but it was slow when highlighting all words based on a list of regexes and the logs were > 10MB. So we tried to optimize every thing we could. Using objects instead of arrays, pagination, reduce the number of regexes. I guess I also switched to the old style for which could make me earn some ms of execution. Maybe after pagination and all other changes this is not necessary anymore but i didn't look at it anymore honestly.

SethFalco commented:
* Moved the config to the centralized config.ts.
* Changed some wording for consistency. (Trying to use the term "Settings" everywhere rather than switch between "Options" or "Settings".
* Add a "Save Settings" button in the webview which saves the 3 settings.

Note, for now this is only allowing `follow`, `timestamp`, and `wrap` to be configured and saved. However, I'll happily do the other settings too, plus the "Run" by default.

Maybe I can do those in a separate PR, though?

I'd like if we can just establish if we're happy with the structure of this first, then I can just follow-through for the rest.

SethFalco commented:
![image](https://user-images.githubusercontent.com/22801583/165944981-19d01b09-a81f-4da7-b29b-1be779914c0e.png)

Ignore the other default values like `0` or `seconds`. (That's for a separate PR I was working on. Just subjective usability improvement and refactors, but that's out of scope for this PR.)

SethFalco commented:
> @SethFalco could you please rebase it? Thanks!!

Rebased and conflicts resolved. 👍🏾

SethFalco commented:
Hmm, the failed test seems odd. 🤔 
Could we maybe do a retry to see if that was a random timeout first?

itowlson commented:
It cure looks like a random timeout - I've re-started the actions.

itowlson reviewed (approved):
LGTM - thanks!

SethFalco commented:
Ahh, sorry about that.
Just a few minutes ago I was looking at this and wanted to tweak `setSettings` a bit, so I force-pushed there anyway. ^-^'

Thought for future, it'd be better if it took an options object rather than a bunch of parameters.

That way in future to set particular settings we could just do:

```js
setSettings({
  sinceSelect: 1
});
```

Instead of something like:
```js
setSettings(null, null, null, 1);
```

lstocchi reviewed (approved):
Wonderful, thanks!! LGTM
